src/vm/class.h

```diff
@@ -375,12 +375,54 @@ class EEClassLayoutInfo
 #endif
 
     private:
+        static void ParseFieldNativeTypes(
```

Comment:
Private static functions are weird in general. They are typically only for consumption of private types - is that the case here? If not, please just put them in the compilation unit.

Reply:
`ParseNativeFieldTypes` and the other ones I add here need access to private members of `EEClassLayoutInfo`. I prefer the private static methods to using friend functions since it helps scope it a little better.

Reply:
Agreed - also not a fan of `friend` functions.